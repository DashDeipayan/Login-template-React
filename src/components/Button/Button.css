.button-div{
    margin-left: 82%;
}
.button{
    padding: 10px 20px;
}