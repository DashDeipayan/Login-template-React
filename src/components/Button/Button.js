import React from "react";
import "./Button.css";

const Button = ({ buttontext }) => {
	return (
		<div className="button-div" data-testid="test-button">
			<button className="button">{buttontext}</button>
		</div>
	);
};

export default Button;
