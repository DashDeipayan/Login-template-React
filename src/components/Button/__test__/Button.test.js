import React from "react";
import ReactDOM from "react-dom";
import Button from "./../Button";

import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import renderer from "react-test-renderer";

afterEach(cleanup);
it("renders without crashing", () => {
	const div = document.createElement("div");
	ReactDOM.render(<Button></Button>, div);
});

it("renders button correctly next", () => {
	const { getByTestId } = render(<Button buttontext="Next"></Button>);
	expect(getByTestId("test-button")).toHaveTextContent("Next");
});

it("renders button correctly submit", () => {
	const { getByTestId } = render(<Button buttontext="Submit"></Button>);
	expect(getByTestId("test-button")).toHaveTextContent("Submit");
});

it("matches snapshot", () => {
	const tree = renderer.create(<Button label="Submit"></Button>).toJSON();
	expect(tree).toMatchSnapshot();
});
