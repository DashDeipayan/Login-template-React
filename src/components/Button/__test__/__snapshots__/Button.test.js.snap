// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="button-div"
  data-testid="button"
>
  <button
    className="button"
  />
</div>
`;
